docs/marvell/build.txt

```diff
@@ -0,0 +1,182 @@
+TF-A Build Instructions
+======================
+
+This section describes how to compile the ARM Trusted Firmware (TF-A) project for Marvell's platforms.
+
+Build Instructions
+------------------
+(1) Set the cross compiler::
+
+		> export CROSS_COMPILE=/path/to/toolchain/aarch64-marvell-linux-gnu-
```

Comment:
Interesting, is that toolchain used in order to optimize the result or is it that ``aarch64-linux-gnu-`` simply doesn't work for you?

Reply:
Oops, this is the old one. We do not use any internally built toolchains. Linaro only now.

Reply:
As for UART download with the current code, I need that access to uninitialized console driver will not cause error or exception. Instead all such attempts should be just ignored.
The BLE and DDR init code will try to output some error messages during image download and I have to ignore this attempts in order to comply with xmodem protocol.

Reply:
I see. so you need a working console for xmodem, but you want to filter all print commands and only let putc from xmodem work, right? I'm thinking that the multi console driver would work.

The way I'd do it is to unregister the console right before the file transfer routine, then use the direct console functions for the file transfer (any printf will be ignored because there is no registered console) and then register the console when the transfer is finished.

Reply:
That is basically what we do. The BLE (BootROM stage code that uses TF-A code as a library and a build environment) is loaded by SoC BootROM into L3 cache, detects that it runs in UART mode and bypasses the console initialization. Then all code that uses TF_A console APIs effectively skipped. If I recall correctly this BLE code was initially based on ATF-v1.2, where the console driver possible ignored "putc" and "getc" when the console base address was 0. So it worked well until asserts were added.
c13b2e32 (Varun Wadekar 2015-05-19 16:44:17 +0530 103) func console_core_putc
c13b2e32 (Varun Wadekar 2015-05-19 16:44:17 +0530 104)  /* Check the input parameter */
c13b2e32 (Varun Wadekar 2015-05-19 16:44:17 +0530 105)  cbz     x1, putc_error

Reply:
Yes, I see, that's why I say that the multi console driver can help. With the multi console driver, ``putc`` can work without any active console, it simply does nothing. However, ``core_xxxx_putc`` can't. Now it is the multi console driver the one that is supposed to have this high level functionality, not the core console functions.

The idea is to move away from the old console in favor of the multi console, so you will have to migrate eventually. Maybe this is a good reason to do it.

Reply:
Sure, I have planned to do this migration. But probably not in this patch series.